Synview-frontend/src/components/DashboardStart.tsx

```diff
@@ -1,9 +1,46 @@
 import React from "react";
+import { useEffect, useState } from "react";
+import { getPayload } from "../Api_handler.ts";
+import { z } from "zod";
+import { useAppSelector, useAppDispatch } from "../hooks.ts";
+import { addUser } from "../slices/userSlice.ts";
 export default function DashboardStart() {
+  type UserInfo = z.infer<typeof UserInfoSchema>;
+  const UserInfoSchema = z.object({
+    username: z.string(),
+    role: z.string(),
+    id: z.number(),
+  });
+
+  const dispatch = useAppDispatch();
+  const user = useAppSelector((state) => state.user);
+
+  const getUserInfo = async () => {
+    let a = await getPayload();
+    try {
+      if (!a) {
+        throw new Error("no payload");
+      }
+      a = UserInfoSchema.parse(a); // <--- For some reason this line isn't necessary but im afraid to remove it
```

Comment:
Your `getPayload` is returning the raw results from calling `.json()` on a request, which is going to be the deserialized object, but with an `any` type. I'd suggest doing the zod validation within your `Api_handler` file itself so you return a typed result there.

Files in TS are typically camelcase unless they are primarily exposing a class or class like thing (React functional component). `apiHandler` would be a better name.